pkg/multicluster/watch/watcher.go

```diff
@@ -88,18 +86,33 @@ func (s *clusterWatcher) ListClusters() []string {
 	return s.managers.list()
 }
 
-func (c *clusterWatcher) startManager(clusterName string, mgr manager.Manager) {
-	ctx, cancel := context.WithCancel(
-		contextutils.WithLoggerValues(context.WithValue(c.ctx, "cluster", clusterName), zap.String("cluster", clusterName)))
-	go func() {
-		err := mgr.Start(ctx)
-		if err != nil {
-			contextutils.LoggerFrom(ctx).DPanicf("manager start failed for cluster %v: %v", clusterName, err)
+func (c *clusterWatcher) startManager(clusterName string, restCfg *rest.Config) {
+	go func() { // this must be async because mgr.Start(ctx) is blocking
+		retryOptions := []retry.Option{
+			retry.Delay(time.Second),
+			retry.Attempts(12),
+			retry.DelayType(retry.BackOffDelay),
 		}
-	}()
 
-	c.managers.set(clusterName, mgr, ctx, cancel)
-	c.handlers.AddCluster(ctx, clusterName, mgr)
+		retry.Do(func() error {
+			ctx, cancel := context.WithCancel(contextutils.WithLoggerValues(context.WithValue(c.ctx, "cluster", clusterName), zap.String("cluster", clusterName)))
+			mgr, _ := manager.New(restCfg, c.optionsWithDefaults())
+
+			// add manager to managers+handlers.  It may fail to start and need to be removed ¯\_(ツ)_/¯
+			c.managers.set(clusterName, mgr, ctx, cancel)
+			c.handlers.AddCluster(ctx, clusterName, mgr)
+
+			err := mgr.Start(ctx) // blocking until error is thrown
```

Comment:
Does this ever return `nil` (like a normal shutdown context)? Should we check before we write a scary error log?